src/archemist/core/persistence/object_factory.py

```diff
@@ -149,13 +149,7 @@ def create_handler(station: Station, use_sim_handler: bool=False):
 
 class OptimizationFactory:
     @staticmethod
-    def create_from_dict(config_dict: Dict) -> OptimizationState:
-        from archemist.core.state.optimisation_state import OptimizationState
-        return OptimizationState.from_dict(config_dict)
+    def create_from_dict(optimizer_type: str, optimization_state, config_dict: Dict ) -> OptimizationState:
+        return optimizer_type(optimization_state, config_dict)
```

Comment:
please change to:
@staticmethod
def create_optimiser_from_dict(optimisation_state: OptimizationState, confg_dict: Dict):
    module = importlib.import_module(confg_dict["module"])
     cls = getattr(config_dict["type"], module)
     return cls(optimisation_state)